test_with_spec_integer_method fails on s390x

**Describe the issue**:

I'm not sure if this is a big-endian problem, or some other thing; I'm not really sure what the test is about.

```pytb
________________________ test_with_spec_integer_method _________________________
[gw2] linux -- Python 3.12.0 /usr/bin/python3
    def test_with_spec_integer_method():
        from dask.dataframe.io.demo import ColumnSpec, DatasetSpec, with_spec
    
        spec = DatasetSpec(
            npartitions=1,
            nrecords=5,
            column_specs=[
                ColumnSpec(prefix="pois", dtype=int, method="poisson"),
                ColumnSpec(prefix="norm", dtype=int, method="normal"),
                ColumnSpec(prefix="unif", dtype=int, method="uniform"),
                ColumnSpec(prefix="binom", dtype=int, method="binomial", args=(100, 0.4)),
                ColumnSpec(prefix="choice", dtype=int, method="choice", args=(10,)),
                ColumnSpec(prefix="rand", dtype=int, random=True, low=0, high=10),
                ColumnSpec(prefix="rand", dtype=int, random=True),
            ],
        )
        ddf = with_spec(spec, seed=42)
        res = ddf.compute()
>       assert res["pois1"].tolist() == [1002, 985, 947, 1003, 1017]
E       assert [1021, 1006, 1024, 1030, 930] == [1002, 985, 947, 1003, 1017]
E         At index 0 diff: 1021 != 1002
E         Full diff:
E         - [1002, 985, 947, 1003, 1017]
E         + [1021, 1006, 1024, 1030, 930]
../../../BUILDROOT/python-dask-2023.7.1-1.fc39~bootstrap.s390x/usr/lib/python3.12/site-packages/dask/dataframe/io/tests/test_demo.py:338: AssertionError
```

**Minimal Complete Verifiable Example**:

```
pytest
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2023.7.1
- Python version: 3.12b4
- Operating System: Fedora Rawhide
- Install method (conda, pip, source): source

--- a/dask/utils.py
+++ b/dask/utils.py
@@ -557,7 +557,7 @@ def random_state_data(n: int, random_state=None) -> list:
         random_state = np.random.RandomState(random_state)
 
     random_data = random_state.bytes(624 * n * 4)  # `n * 624` 32-bit integers
-    l = list(np.frombuffer(random_data, dtype=np.uint32).reshape((n, -1)))
+    l = list(np.frombuffer(random_data, dtype="<u4").reshape((n, -1)))
     assert len(l) == n
     return l
 

